### ISSUE TITLE
Type guard of Number.isNaN is wrong

### ISSUE THREAD

**Issue by SaschaNaz** [Microsoft/TypeScript#12302]
**TypeScript Version:** 2.1 RC bundled in VS2017 RC

**Code**

```ts
let num = 8;
if (!Number.isNaN(num)) { // NumberConstructor.isNaN(value: any): value is number
    printNumber(num + 1); // `num` should still be a number but TS thinks it has `never` type
}

function printNumber(num: number) {
    console.log(`Got ${num} - 1`);
}
```

**Expected behavior:** No error

**Actual behavior:** TS2356 An arithmetic operand must be of type 'any', 'number' or an enum type.

**Comment by akarzazi** [comment]
This code would raise an other error on `num++` since it was declared with `const`.

Javascript :
```
const num = 0;
num++;
```
`VM173:2 Uncaught TypeError: Assignment to constant variable`

**Comment by RyanCavanaugh** [comment]
Introduced via #11670. Seemed like a good idea at the time!
